libdnf/rpm/solv/goal_private.hpp

```diff
@@ -115,7 +115,7 @@ class GoalPrivate {
     bool remove_solver_weak{false};
 
     /// Return true when protected packages including kernel is in removal or obsoleted list
-    bool protected_in_removals();
+    libdnf::GoalProblem protected_in_removals();
```

Comment:
Does not return "true" as documented line above.
Are you planning more states in the future, or will the function return only two states?  - Are they protected during removal? - `true` or` false`?

Reply:
Correct, I change it because I like in libdnf::GoalProblem GoalPrivate::resolve() { to `return protected_in_removals();`. But I can change it again. Also I had a plan to extend possibility to not only report  REMOVAL_OF_PROTECTED, but also removal of running kernel. But I did not implement it because I did not know if it is required. @jrohel What do you prefer?